project/src/main/java/com/google/sps/servlets/BlobFormHandlerServlet.java

```diff
@@ -0,0 +1,90 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobInfoFactory;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.images.ImagesService;
+import com.google.appengine.api.images.ImagesServiceFactory;
+import com.google.appengine.api.images.ServingUrlOptions;
+import com.google.sps.data.User;
+import com.google.sps.utilities.SpannerTasks;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * When the user submits the form, Blobstore processes the file upload and then forwards the request
+ * to this servlet. This servlet can then process the request using the file URL we get from
+ * Blobstore.
+ */
+@WebServlet("/blob-handler")
+public class BlobFormHandlerServlet extends HttpServlet {
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Optional<User> userOptional = SpannerTasks.getLoggedInUser();
+    if (userOptional.isPresent()) {
+      User user = userOptional.get();
+      response.getWriter().println(user.getImageUrl());
+    } else {
+      response.sendError(
+          HttpServletResponse.SC_BAD_REQUEST, "Error with getting current user: does not exist");
```

Comment:
You could use the return-early pattern here
if (!userOptional.isPresent()) {
   sendError
   return
}

send response
I have tried to follow this pattern but it is up to you.

Reply:
Ack, that does make sense - I think personally, it reads easier to me as "if userOptional is present... otherwise" instead of the negated one first